<?xml version="1.0"?>
<launch>
    <!-- <args for urdf/> -->
    <arg name="camera_model"          default="OAK-D"   /> <!-- 'zed' or 'zedm' -->
    <arg name="tf_prefix"             default="oak"         />
    <arg name="base_frame"            default="oak-d_frame" />
    <arg name="parent_frame"          default="oak-d-base-frame" />
    <arg name="publisher_name"        default="tracker_publisher" />

    <arg name="cam_pos_x"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"              default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch"             default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw"               default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->

    <arg name="camera_param_uri"      default="package://depthai_examples/params/camera" />
    <arg name="sync_nn"               default="true"/>
    <arg name="subpixel"              default="true"/>
    <arg name="confidence"            default="200" /> 
    <arg name="LRchecktresh"          default="5"   /> 
    <arg name="lrcheck"               default="true" /> 
    <arg name="extended"              default="false" /> 
    <arg name="monoResolution"        default="400p"/> <!-- '720p', '800p', 400p' for OAK-D & '480p' for OAK-D-Lite -->
    <arg name="monoFPS"               default="30"/>
    <arg name="rgbFPS"                default="30"/>
    <arg name="nnName"                default="x" />
    <arg name="resourceBaseFolder"    default="$(find depthai_examples)/resources" />
    <arg name="previewSize"           default="416" />
    <arg name="rgbResolution"         default="1080p"/>
    <arg name="rgbScaleNumerator"     default="2"/>
    <arg name="rgbScaleDinominator"   default="3"/>
    <arg name="enableDotProjector"    default="true"/>
    <arg name="enableFloodLight"      default="true"/>
    <arg name="dotProjectormA"        default="1200.0"/>
    <arg name="floodLightmA"          default="400.0"/>
    <arg name="angularVelCovariance"  default="0"   />
    <arg name="linearAccelCovariance" default="0"   />

    <include file="$(find depthai_descriptions)/launch/urdf.launch">
        <arg name="base_frame"      value="$(arg  base_frame)"  />
        <arg name="parent_frame"    value="$(arg  parent_frame)"/>
        <arg name="camera_model"    value="$(arg  camera_model)"/>
        <arg name="tf_prefix"       value="$(arg  tf_prefix)" />
        <arg name="cam_pos_x"       value="$(arg  cam_pos_x)" />
        <arg name="cam_pos_y"       value="$(arg  cam_pos_y)" />
        <arg name="cam_pos_z"       value="$(arg  cam_pos_z)" />
        <arg name="cam_roll"        value="$(arg  cam_roll)"  />
        <arg name="cam_pitch"       value="$(arg  cam_pitch)" />
        <arg name="cam_yaw"         value="$(arg  cam_yaw)"   />
    </include>

    <node name="$(arg publisher_name)" pkg="depthai_examples" type="tracker_node" output="screen" required="true">
        <param name="tf_prefix"           value="$(arg tf_prefix)"     />
        <param name="camera_param_uri"    value="$(arg camera_param_uri)"/>
        <param name="sync_nn"             value="$(arg sync_nn)"/>
        <param name="subpixel"            value="$(arg subpixel)"/>
        <param name="confidence"          value="$(arg confidence)"/>
        <param name="LRchecktresh"        value="$(arg LRchecktresh)"/>
        <param name="monoResolution"      value="$(arg monoResolution)"/>
        <param name="monoFPS"             value="$(arg monoFPS)"/>
        <param name="nnName"              value="$(arg nnName)"/>
        <param name="resourceBaseFolder"  value="$(arg resourceBaseFolder)"/>
        <param name="previewSize"         value="$(arg previewSize)"/>
        <param name="lrcheck"             value="$(arg lrcheck)"/> 
        <param name="extended"            value="$(arg extended)"/> 
        <param name="rgbFPS"              value="$(arg rgbFPS)"/> 
        <param name="rgbResolution"       value="$(arg rgbResolution)"/> 
        <param name="rgbScaleNumerator"   value="$(arg rgbScaleNumerator)"/> 
        <param name="rgbScaleDinominator" value="$(arg rgbScaleDinominator)"/> 
        <param name="enableDotProjector"  value="$(arg enableDotProjector)"/> 
        <param name="enableFloodLight"    value="$(arg enableFloodLight)"/> 
        <param name="dotProjectormA"      value="$(arg dotProjectormA)"/> 
        <param name="floodLightmA"        value="$(arg floodLightmA)"/> 
        <param name="angularVelCovariance"  value="$(arg angularVelCovariance)"   />
        <param name="linearAccelCovariance" value="$(arg linearAccelCovariance)"   />
    </node>

</launch>
